cli_tools/gce_windows_upgrade/upgrader/utils.go

```diff
@@ -0,0 +1,146 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package upgrader
+
+import (
+	"bytes"
+	"strings"
+	"text/template"
+
+	daisyutils "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/daisy"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/param"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/daisycommon"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+)
+
+const (
+	upgradeIntroductionTemplate = "The following resources will be created/accessed during the upgrade. " +
```

Comment:
I removed "before you begin the upgrade " because the upgrade has started when user see this text;
I removed "after a failed upgrade" because cleanup resources can be done after a success upgrade. User may also do manual rollback after a success upgrade inc ase they regret.

Reply:
SGTM.